attest/ake/src/initiator.rs

```diff
@@ -152,54 +150,54 @@ where
-                let mut verifier = Verifier::default();
-                verifier.identities(&identities).debug(DEBUG_ENCLAVE);
-
-                // We are not returning the report data and instead returning the raw report
-                // since that also includes the signature and certificate chain.
-                // However, we still make sure the report contains valid data
-                // before we continue by calling `.verify`. Callers can then
-                // safely construct a VerificationReportData object out of the
-                // VerificationReport returned.
-                let _report_data = verifier
-                    .report_data(
-                        &result
-                            .remote_identity
-                            .ok_or(Error::MissingRemoteIdentity)?
-                            .map_bytes(|bytes| {
-                                ReportDataMask::try_from(bytes)
-                                    .map_err(|_| Error::BadRemoteIdentity)
-                            })?,
-                    )
-                    .verify(&remote_report)?;
-                Ok((
-                    Ready {
-                        writer: result.initiator_cipher,
-                        reader: result.responder_cipher,
-                        binding: result.channel_binding,
-                    },
-                    remote_report,
-                ))
+                let remote_identity = result
+                    .remote_identity
+                    .as_ref()
+                    .ok_or(Error::MissingRemoteIdentity)?;
+                if let Ok(remote_evidence) = EvidenceKind::from_bytes(output.payload.as_slice()) {
+                    verify_evidence_kind(
+                        remote_evidence.clone(),
+                        &input.identities,
+                        input.time,
+                        remote_identity,
+                    )?;
+                    Ok((
+                        Ready {
+                            writer: result.initiator_cipher,
+                            reader: result.responder_cipher,
+                            binding: result.channel_binding,
+                        },
+                        remote_evidence,
+                    ))
+                } else {
+                    let remote_report = VerificationReport::decode(output.payload.as_slice())
+                        .map_err(|_| Error::AttestationEvidenceDeserialization)?;
+                    verify_verification_report(&remote_report, &input.identities, remote_identity)?;
+                    Ok((
+                        Ready {
+                            writer: result.initiator_cipher,
+                            reader: result.responder_cipher,
+                            binding: result.channel_binding,
+                        },
+                        EvidenceKind::Epid(remote_report),
+                    ))
+                }
```

Comment:
I think this can be reduced to 
```suggestion
                let remote_evidence = EvidenceKind::from_bytes(output.payload.as_slice())
                    .map_err(|_| Error::AttestationEvidenceDeserialization)?;
                verify_evidence_kind(
                    remote_evidence.clone(),
                    &input.identities,
                    input.time,
                    remote_identity,
                )?;
                Ok((
                    Ready {
                        writer: result.initiator_cipher,
                        reader: result.responder_cipher,
                        binding: result.channel_binding,
                    },
                    remote_evidence,
                ))
```